packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -245,6 +245,24 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
     setTextValue(undefined);
   };
 
+  let valueToDisplay;
+  if (textValue !== undefined) {
+    valueToDisplay = textValue;
+  } else if (value === null || currentValue === null) {
+    valueToDisplay = displayValue ?? '';
+    internalState.current.value = null;
+    internalState.current.atBound = 'none';
+  } else {
+    const roundedValue = precisionRound(currentValue, precision);
+    internalState.current.value = roundedValue;
+    internalState.current.atBound = getBound(roundedValue, min, max);
```

Comment:
The change is that we need to set this internal state before setting the `SpinButtonState` below.